subprojects/workers/src/main/java/org/gradle/workers/BaseWorkerSpec.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.workers;
+
+import org.gradle.api.Action;
+import org.gradle.api.Describable;
+import org.gradle.api.Incubating;
+import org.gradle.process.JavaForkOptions;
+
+import java.io.File;
+
+/**
+ * Represents the common configuration of a worker.  Used when submitting an item of work
+ * to the {@link WorkerExecutor}.
+ *
+ * @since 5.6
+ */
+@Incubating
+public interface BaseWorkerSpec extends Describable {
+    /**
+     * Adds a set of files to the classpath associated with the worker.
+     *
+     * @param files - the files to add to the classpath
+     */
+    void classpath(Iterable<File> files);
+
+    /**
+     * Sets the classpath associated with the worker.
+     *
+     * @param files - the files to set the classpath to
+     */
+    void setClasspath(Iterable<File> files);
+
+    /**
+     * Gets the classpath associated with the worker.
+     *
+     * @return the classpath associated with the worker
+     */
+    Iterable<File> getClasspath();
```

Comment:
Maybe we should consider keeping these methods in `WorkerConfiguration` and using a `ConfigurableFileCollection` in `WorkerSpec` instead.  

The migration path would be:

```
workerExecutor.submit(MyExecution) {  config -> 
   config.classpath myExtraClasspath
}
```
to
```
workerExecutor.newSubmitMethod(MyExecution) {  config -> 
   config.classpath.from(myExtraClasspath)
}
```

Reply:
I like it.  Done.